pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -131,11 +140,16 @@ func (a *index) constructServiceEntries(svc *networkingclient.ServiceEntry, w *W
 	// TODO this is only checking one controller - we may be missing service vips for instances in another cluster
 	res := make([]*workloadapi.Service, 0, len(svc.Spec.Hosts))
 	for _, h := range svc.Spec.Hosts {
+		// if we have no user-provided addresses and h is not wildcarded we can try to use autoassigned addresses
```

Comment:
I don't think a SE with an explicit user address and one with auto-allocated address should be in any way different ( assuming we do add the selective enablement - so users can continue having SE that don't get assigned addresses that user doesn't need/want)

Reply:
We only have any addresses in the `autoassignedAddresses` variable if `serviceentry.ShouldV2AutoAllocateIP` returns true so if user given addresses are empty and we didn't have auto allocate on then the result will be an empty list of addresses. I think that is what we want, yeah?

Reply:
Well - not entirely. Other controllers ( other Istiod revision or something external ) may assign IP addresses as well, so 'ShouldV2Autoallocate' or whatever labels we have are only about deciding if THIS istiod instance should allocate the addresses. 

We should make it very clear in the docs for the enablement. 

If neither 'autoallocate' ( by this Istiod or someone else ) nor the user-specified address are set - the result is an empty list of addresses, but that is split in 2 cases: 
- pure backend ( the SE defines a route destination, user does not intend to use it as a frontend and capture). I don't know how this is handled by Ambient - I assume a TcpRoute will be set from a FE/VIP and SE is treated like WorkloadEntries.

- HTTP or SNI - where further processing is done at L7 or by parsing SNI, which is a pretty important and tricky feature in Istio. I assume those can't be supported by Ambient - or may trigger a redirect to a gateway/waypoint/egress.

Reply:
I was probably not that clear in my explanation. What I mean is in the ambient controller if `serviceentry.ShouldV2AutoAllocateIP` doesn't return true then we don't bother to get the addresses out of status, even if they exist. We skip that and leave autoassignedAddresses as empty. Here is that relevant bit: https://github.com/istio/istio/blob/42703a09223e4265b71e9b988d7807cf0ed7dbcd/pilot/pkg/serviceregistry/kube/controller/ambient/services.go#L116